Cython/Compiler/ExprNodes.py

```diff
@@ -2645,6 +2726,35 @@ def generate_assignment_code(self, rhs, code, overloaded_assignment=False,
 
             rhs.free_temps(code)
 
+        self.generate_threadsafe_unlock(code)
+
+    def generate_threadsafe_lock(self, code):
+        if not self.needs_threadsafe_access:
+            return
+        # We don't have an obj so I don't see how we can lock
+        assert not self.threadsafety_scope.is_c_class_scope
+        if self.threadsafety_scope.is_closure_scope:
+            scope_cname = self.entry.cname.rsplit('->', 1)[0]
```

Comment:
I'm aware that this is awful. It's basically how we generate these names in symtab though so I couldn't immediately see a better way

Reply:
Maybe add a property `plain_cname` to `Entry`that unpacks such prefixes if necessary? Both `*.x` and `*->x`? That would keep the knowledge about cnames more local.